Fix failing tests on master

We're getting [build failures on `master`](https://github.com/asdf-community/asdf-hashicorp/runs/5993356952?check_suite_focus=true) related to TF `1.2.0-alpha`:

```
Run make test
bats test
1..18
ok 1 install command fails if the input is not version number
ok 2 can install and verify earliest terraform
not ok 3 can install and verify latest terraform
# (in test file test/install.bats, line 31)
#   `echo "$output" | grep "Extracting terraform archive"' failed
# installing terraform 1.2.0-alpha
ok 4 can install and verify earliest vault
ok 5 can install and verify latest vault
ok 6 can install and verify earliest consul
ok 7 can install and verify latest consul
ok 8 supports legacy terraform version 'required_version' with strict equality
ok 9 supports alternate file for terraform version constraints
ok 10 supports legacy terraform version 'required_version' with strict equality, no equals literal
ok 11 supports legacy file .terraform-version
ok 12 does not support 'not equal' version constraint expressions
ok 13 does not support 'greater than' version constraint expressions
ok 14 does not support 'less than' version constraint expressions
ok 15 does not support squiggly arrow version constraint expressions
ok 16 does not support compound version constraint expressions
ok 17 does not get confused by multiple legacy version files for different plugins
ok 18 does not output error if required_version is not specified
make: *** [Makefile:17: test] Error 1
Error: Process completed with exit code 2.
```

Need to debug and resolve (or skip it this is irrelevant) so we can test PRs again.

--- a/bin/list-all
+++ b/bin/list-all
@@ -14,7 +14,7 @@ list_all() {
   local v
 
   for v in $(curl -s "https://releases.hashicorp.com/${toolname}/" |
-    grep -o "${toolname}_[0-9]\+\.[0-9]\+\.[0-9]\+\(\(-\|+\)[a-z]\+[0-9]*\)*"); do
+    grep -o "${toolname}_[0-9]\+\.[0-9]\+\.[0-9]\+\(\(-\|+\)[a-z]\+-\?[0-9]*\)*"); do
     version="${v#"${toolname}"_}"
     if [[ -z ${versions} ]]; then
       versions="${version}"

